tools/contrib/subset_data.py

```diff
@@ -0,0 +1,1096 @@
+                    action="store",   
+                    dest="create_surfdata",    
+                    required = False, 
+                    type = bool,      
+                    default = True)   
+        rg_parser.add_argument('--create_landuse',
+                    help='Flag for creating landuse data file for a region. [default: %(default)s]', 
+                    action="store_true",   
+                    dest="create_landuse",     
+                    required = False, 
+                    default = False)   
+        rg_parser.add_argument('--create_datm',
+                    help='Flag for creating DATM forcing data for a region. [default: %(default)s]', 
+                    action="store_true",   
+                    dest="create_datm",        
+                    required = False, 
+                    default = False)  
+        rg_parser.add_argument('--datm_syr',   
+                    help='Start year for creating DATM forcing for a region. [default: %(default)s]', 
+                    action="store",   
+                    dest="datm_syr",  
+                    required = False, 
+                    type = int,       
+                    default = 1901)
+        rg_parser.add_argument('--datm_eyr',
+                    help='End year for creating DATM forcing for a region.  [default: %(default)s]', 
+                    action="store",
+                    dest="datm_eyr",
+                    required = False,
+                    type = int,
+                    default = 2014)
+        rg_parser.add_argument('--crop', 
+                    help='Create datasets using the extensive list of prognostic crop types. [default: %(default)s]', 
+                    action="store_true", 
+                    dest="crop_flag", 
+                    default=False)
+        rg_parser.add_argument('--dompft', 
+                    help='Dominant PFT type . [default: %(default)s] ', 
+                    action="store", 
+                    dest="dom_pft",
+                    type =int,
+                    default=7)
+        rg_parser.add_argument('--outdir', 
+                    help='Output directory. [default: %(default)s]', 
+                    action="store", 
+                    dest="out_dir",
+                    type =str,
+                    default="/glade/scratch/"+myname+"/regional/")
+
+        return parser
+
+def plat_type(x):
+    """
+    Function to define lat type for the parser
+    and
+    raise error if latitude is not between -90 and 90.
+    """
+    x = float(x)
+    if (x < -90) or (x > 90):
+        raise argparse.ArgumentTypeError("ERROR: Latitude should be between -90 and 90.")
+    return x
+
+
+def plon_type(x):
+    """
+    Function to define lon type for the parser and
+    convert negative longitudes and 
+    raise error if lon is not between -180 and 360.
+    """
+    x = float(x)
+    if (-180 < x) and (x < 0):
+        x= x%360
```

Comment:
It's modulus, remainder after division by 360.

Reply:
is this what generating the difference between LON values in the NIWO file? https://github.com/ESCOMP/CTSM/pull/1278#issuecomment-830451156
If so, this will be an issue for every NEON site, as LON is provided in deg West (e.g. -105).

Reply:
@wwieder 
Thanks for your comment. As @ekluzek mentioned the `%` sign is the remainder of dividing the left hand operand by right hand operand.

Here I used `%` sign to convert lons between -180 to 0 to positive lons. For example:

```
>>> x = -150
>>> print (x%360)
210
```

Reply:
> is this what generating the difference between LON values in the NIWO file? [#1278 (comment)](https://github.com/ESCOMP/CTSM/pull/1278#issuecomment-830451156)
> If so, this will be an issue for every NEON site, as LON is provided in deg West (e.g. -105).

No this is not the reason causing the discrepancy between the NIWO files lon values. 
```
>>> x= 254.41762999999997
>>> print (x%360)
254.41763
```

That is not essentially an error or a bug and it is simply caused by the fact that we are selecting the closest point in the subset_data.py and we are not extrapolating:

```
        # extract gridcell closest to plon/plat                                                                                                
        f3 = f2.sel(ni=self.plon,nj=self.plat,method='nearest')
```

This is similar to the ./singlpt script were we are using the nearest and using that script I get a similar discrepancy.
I will explain further in the PR #1278.

Reply:
should we just write plat and plon the the lat and lon coords of the new single point surface data file, instead of the nearest lat-lon from f2?

Reply:
Yes, you should definitely change the dataset to use the lat/lon of the tower site, since that's the location you are trying to simulate.

Reply:
Actually, there might still be two cases here. If you want to run for a specific tower site you want the lat and lon of the tower site. But, if you just want to run one of the model points as a single point -- you want the lat/lon of that point. This might be similar to the issue about 3-file atm forcing vs. single file for tower site. We might want a flag to throw or examine to decide between these two.